src/java.base/share/classes/sun/nio/cs/StreamEncoder.java

```diff
@@ -106,33 +107,59 @@ public String getEncoding() {
     }
 
     public void flushBuffer() throws IOException {
-        synchronized (lock) {
-            if (isOpen())
-                implFlushBuffer();
-            else
-                throw new IOException("Stream closed");
+        if (lock instanceof InternalLock locker) {
```

Comment:
now that StreamEncoder is final, we can drop the `else` branch

Reply:
Actually, we can’t in case `InternalLock.CAN_USE_INTERNAL_LOCK` is ever `false` (which it might be if it gets exposed as a system property).

Reply:
> Actually, we can’t in case `InternalLock.CAN_USE_INTERNAL_LOCK` is ever `false` 

That's right, both StreamEncoder and StreamDecoder need the both cases.